app/views/restaurants/show.html.haml

```diff
@@ -27,4 +27,5 @@
     $
   %img{alt: dish.name, src: dish.pic_url}/
 
-  = link_to "Add #{dish.name} to cart" , add_to_cart_path(dish, restaurant_id: @restaurant.id)
+  = link_to "Add #{dish.name} to cart", add_to_cart_path(dish, restaurant_id: @restaurant.id)
+  = link_to "Remove #{dish.name} from cart", cart_remove_item_path(dish, restaurant_id: @restaurant.id, dish_id: dish.id)
```

Comment:
Could we only show the remove button if the dish is actually in the cart?

Reply:
yes, will add that.